backend/de.metas.manufacturing.rest-api/src/main/java/de/metas/manufacturing/job/service/commands/ReceiveGoodsCommand.java

```diff
@@ -46,50 +56,70 @@ public class ReceiveGoodsCommand
 	//
 	// Services
 	private final IHandlingUnitsBL handlingUnitsBL;
+	private final IUOMConversionBL uomConversionBL;
+	private final IUOMDAO uomDao;
 	private final IHUPPOrderBL ppOrderBL;
 	private final IPPOrderBOMBL ppOrderBOMBL;
+	private final IHUPIItemProductBL huPIItemProductBL;
 	private final ManufacturingJobLoaderAndSaverSupportingServices loadingAndSavingSupportServices;
 
 	//
 	// Parameters
 	final @NonNull PPOrderId ppOrderId;
-	final @Nullable PPOrderBOMLineId coProductBOMLineId;
+	final @NonNull FinishedGoodsReceiveLine receiveLine;
 	final @NonNull SelectedReceivingTarget receivingTarget;
 	final @NonNull BigDecimal qtyToReceiveBD;
 	final @NonNull ZonedDateTime date;
+	final @Nullable String bestBeforeDate;
+	final @Nullable String lotNo;
+	final @Nullable Quantity catchWeight;
 
 	//
 	// State
 	private I_PP_Order _ppOrder; // lazy
 	private I_PP_Order_BOMLine _coProductLine;
+	private final List<I_M_HU> receivedHUs = new ArrayList<>();
 
 	@Builder
 	private ReceiveGoodsCommand(
 			@NonNull final IHandlingUnitsBL handlingUnitsBL,
+			@NonNull final IUOMConversionBL uomConversionBL,
+			@NonNull final IUOMDAO uomDao,
 			@NonNull final IHUPPOrderBL ppOrderBL,
 			@NonNull final IPPOrderBOMBL ppOrderBOMBL,
+			@NonNull final IHUPIItemProductBL huPIItemProductBL,
 			@NonNull final ManufacturingJobLoaderAndSaverSupportingServices loadingAndSavingSupportServices,
 			//
 			@NonNull final PPOrderId ppOrderId,
-			@Nullable final PPOrderBOMLineId coProductBOMLineId,
+			@NonNull final FinishedGoodsReceiveLine receiveLine,
 			@NonNull final SelectedReceivingTarget receivingTarget,
 			@NonNull final BigDecimal qtyToReceiveBD,
-			@NonNull final ZonedDateTime date)
+			@NonNull final ZonedDateTime date,
+			@Nullable final String bestBeforeDate,
+			@Nullable final String lotNo,
+			@Nullable final BigDecimal catchWeight,
```

Comment:
use Quantity

Reply:
hmmm or... i think we can keep it as `BigDecimal catchWeight` but them we shall not take the UOM from frontend because we know that's the catch weight UOM, don't we?